components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponentHolder.java

```diff
@@ -66,4 +71,32 @@ public static RegistryService getRegistryService() {
     public static void setRegistryService(RegistryService registryService) {
         OAuth2ServiceComponentHolder.registryService = registryService;
     }
+
+    /**
+     * Get ClaimProvider Service
+     *
+     * @return all ID token claims
+     */
+    public static List<ClaimProvider> getClaimProviders() {
+        return claimProviders;
+    }
+
+    /**
+     * Set ClaimProvider Service
+     *
+     * @param claimProvider
+     */
+    public static void setClaimProvider(ClaimProvider claimProvider) {
```

Comment:
setClaimProvider and unregisterClaimProvider are not consistent naming IMO.
Moreover do we need these two methods?

Why not use OAuth2ServiceComponentHolder.getInstance().getClaimProviders().add() and 
OAuth2ServiceComponentHolder.getInstance().getClaimProviders().remove()  instead

Reply:
Fixed